src/utils/parser.js

```diff
@@ -181,6 +198,31 @@ function getSymbols(source: SourceText): SymbolDeclarations {
   return symbols;
 }
 
+function getExpression(source: SourceText, token: string, location: Location) {
+  let expression = null;
+  const ast = getAst(source);
+
+  if (isEmpty(ast)) {
+    return;
+  }
+
+  traverse(ast, {
```

Comment:
Maybe it'd be a bit more efficient if we had a single function that did traversal? I know `getSymbols` does it and so does `getPathClosestToLocation` we could limit our traversals of documents to single places and return a single `Map` with all the relevant goodies.